tests/5.1/teams/test_target_set_num_teams.c

```diff
@@ -0,0 +1,52 @@
+//===--------------------- test_target_set_num_teams.c ----------------------===//
+//
+// OpenMP API Version 5.1 Nov 2020
+//
+// This test uses the omp_set_num_teams utility to set the default number of
+// teams allocated for team clauses. The test runs teams and checks to see if
+// the number allocated is correct and reflective of the new default set. It
+// then checks to see if an explicitly stated num_teams clause overrides
+// the omp_set_num_teams properly.
+//
+//===------------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+
+int main() {
+
+	OMPVV_TEST_OFFLOADING;
+
+	int errors = 0; 
+	int num_teams = 0;
+
+        #pragma omp target
+        {
+	  omp_set_num_teams(4);
+        }
+	
+	#pragma omp target teams map(tofrom: num_teams)
+	{
+		if (omp_get_team_num() == 0 ) {
+			num_teams = omp_get_num_teams();
+		}
+	}               
+	
+	OMPVV_ERROR_IF(num_teams != 4, "Number of teams detected was not the number set by omp_set_num_teams()");
+	OMPVV_TEST_AND_SET(errors, num_teams != 4);
+
+	#pragma omp target teams map(tofrom: num_teams) num_teams(OMPVV_NUM_TEAMS_DEVICE)
```

Comment:
This test is not concerned with testing the num_teams clause as the num_teams clause does not set the nteams ICV which omp_set_num_teams() relates to. As such, lines 40-49 can possibly be removed.

Reply:
This test does not use the method to retrieve the nteams ICV, omp_get_max_teams(). So the actual value of the variable is not being queried. The number of teams created in the teams region (with no clause) is being queried, however it is being expected that the number queried is equal to the number set, when it is legal to have the value be less than or equal to the number set.